seqr/views/apis/family_api_tests.py

```diff
@@ -59,48 +98,77 @@ def test_update_family_pedigree_image(self):
         self.assertEqual(response.status_code, 200)
         response_json = response.json()
 
-        self.assertListEqual(response_json.keys(), ['F000001_1'])
-        self.assertEqual(response_json['F000001_1']['assignedAnalyst']['email'], 'test_user@test.com')
-        self.assertEqual(response_json['F000001_1']['assignedAnalyst']['fullName'], 'Test User')
+        self.assertListEqual(response_json.keys(), [FAMILY_GUID])
+        self.assertEqual(response_json[FAMILY_GUID]['assignedAnalyst']['email'], 'test_user@test.com')
+        self.assertEqual(response_json[FAMILY_GUID]['assignedAnalyst']['fullName'], 'Test User')
 
         # unassign analyst
         response = self.client.post(url, content_type='application/json', data=json.dumps({}))
         self.assertEqual(response.status_code, 200)
         response_json = response.json()
-        self.assertListEqual(response_json.keys(), ['F000001_1'])
-        self.assertIsNone(response_json['F000001_1']['assignedAnalyst'])
+        self.assertListEqual(response_json.keys(), [FAMILY_GUID])
+        self.assertIsNone(response_json[FAMILY_GUID]['assignedAnalyst'])
 
     def test_update_success_story_types(self):
         url = reverse(update_family_fields_handler, args=[FAMILY_GUID])
         _check_login(self, url)
 
         # send valid request
-        response = self.client.post(url, content_type='application/json', data=json.dumps({'successStoryTypes': ['O', 'D']}))
+        response = self.client.post(url, content_type='application/json',
+                                    data=json.dumps({'successStoryTypes': ['O', 'D']}))
         self.assertEqual(response.status_code, 200)
         response_json = response.json()
-        self.assertListEqual(response_json['F000001_1']['successStoryTypes'], ['O', 'D'])
+        self.assertListEqual(response_json[FAMILY_GUID]['successStoryTypes'], ['O', 'D'])
+
+    def test_receive_families_table_handler(self):
+        url = reverse(receive_families_table_handler, args=[PROJECT_GUID])
+        _check_login(self, url)
+
+        # send request with a "families" attribute
+        data = b'Family ID	Display Name	Description	Coded Phenotype\n\
+"1"	"1"	"sf"	"LEFT VENTRICULAR NONCOMPACTION 10; LVNC10"\n\
+"2"	"2"	"sz test"	""'
+
+        f = SimpleUploadedFile("1000_genomes demo_families.tsv", data)
+
+        response = self.client.post(url, {'f': f})
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+
+        self.assertListEqual(response_json.keys(), ['info', 'errors', 'warnings', 'uploadedFileId'])
+
+        url = reverse(edit_families_handler, args=[PROJECT_GUID])
+
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'uploadedFileId': response_json['uploadedFileId']}))
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+
+        self.assertListEqual(response_json.keys(), ['familiesByGuid'])
```

Comment:
Can you please add some tests that the family data here is correct

Reply:
Response data checking is added.